lib/rspec/core/configuration.rb

```diff
@@ -1412,13 +1415,32 @@ def self.delegate_to_ordering_manager(*methods)
       #     end
       #   end
       #
-      #   describe MyClass, :order => :reverse do
+      #   RSpec.describe 'MyClass', :order => :reverse do
       #     # ...
       #   end
       #
       # @note Pass the symbol `:global` to set the ordering strategy that
       #   will be used to order the top-level example groups and any example
       #   groups that do not have declared `:order` metadata.
+      #
+      # @example
+      #   RSpec.configure do |rspec|
+      #     rspec.register_ordering :global do |examples|
+      #       acceptance, other = examples.partition do |example|
+      #         example.metadata[:type] == :acceptance
```

Comment:
The most common answer to "what do you want?" is: "it depends on what's possible".

The example has to be no only clear, but compelling. If it's compelling and useful, people won't mind if it's complex.

If complexity is problem, the API has to change. E.g. it's not "obvious" that first example groups are sorted using the strategy, and then individual examples within the group are sorted with the same strategy. For example, this means example group metadata (e.g. filenames -`file_path`) are only useful the first time the method is invoked. So the user may want to distinguish between "sorting groups" vs "sorting examples within a group".

Also, the fact that the user can filter out examples during ordering isn't obvious either. The whole concept here seems like exposing RSpec internals to the user. (Instead of just being able to set 2 comparison methods: one for examples and one for groups).

Ruby's docs with `sort` examples are trivial compared to this. Even though the idea is the same.